native: fix 1 config warning in native

aescolar opened the pull request:
<del>
Two minor fixes:

1. Let samples/hello_world/prj_single be run in boards
which by default compile BT by unselecting it in the
prj configuration.
</del>

2. The native SOC Kconfig file was still referring to the old
configuration option SOC_INF_CLOCK which was replaced with
SOC_POSIX.
This produced a warning on configuration which is now fixed

3. Fixed name of the board in config option short description

Signed-off-by: Alberto Escolar Piedras <alpi@oticon.com>

aescolar committed:
native: fixed configuration warning

The native SOC Kconfig file was still referring to the old
configuration option SOC_INF_CLOCK which was replaced with
SOC_POSIX.
This produced a warning on configuration which is now fixed

Signed-off-by: Alberto Escolar Piedras <alpi@oticon.com>

SebastianBoe commented:
I don't agree with the approach in the first fix.

The hello world sample should not have to turn off features it should not need to know about.

Instead, the boards that are defining BT=y should stop doing so. Resolving this issue is the correct approach to the first issue IMHO: https://github.com/zephyrproject-rtos/zephyr/issues/5454

SebastianBoe reviewed (changes requested):
See comment

aescolar-ot commented:
> I don't agree with the approach in the first fix.

@SebastianBoe: I change this PR to only include the warning fix then. I let somebody else take the hello_word CONFIG_BT issue if they want to

aescolar committed:
native: Corrected board name in config short description

Corrected the board name in the configuration short
description string. It was the old Simple_process name which was
replaced with native_posix

Corrected also the comment/title in the "Board Options" menu

Signed-off-by: Alberto Escolar Piedras <alpi@oticon.com>

nashif commented on boards/posix/native_posix/Kconfig.board:
```diff
@@ -1,6 +1,6 @@
 
 config BOARD_NATIVE_POSIX
-	bool "Posix simple process"
+	bool "native_posix board"
```
Should be readable format, i.e. "Native Posix"
